weave-js/src/common/util/links.tsx

```diff
@@ -115,6 +112,18 @@ export const TargetBlank: FCWithRef<
   })
 );
 
+export const getAbsolutePrefixedUrl = (href?: string) => {
+  if (href == null) {
+    return undefined;
+  }
+
+  if (href.indexOf('mailto:') === 0 || href.indexOf('://') > 0) {
```

Comment:
i'm curious if there's a way to embed multiple schemes, but this is probably good assuming security likes it. I'd generally prefer using the real URL parser if we need to do anything more complex in security settings, just because complicated things like URLs can do some crazy stuff.

Reply:
Are we expecting the source of URLs to come from end users? If so, to your point Stephen, we will need a security check at the end of all the Url processing like [so](https://github.com/wandb/core/blob/b241247e5a173d1e4ad29d3f8b4f59dc4ade8a19/frontends/app/src/util/report/urls.ts#L55) to avoid XSS. Maybe we can add `mailto:` to the linked security check?

The risk here is that if we pass an href like `javascript:alert('cookies'+'://')`, this should return `javascript:alert('cookies'+'://')` as is, assuming the host is not prefixed. We'll get something like the below anchor element which would result in stored XSS. Opening the below html file and command+clicking the link should trigger the javascript. 

index.html
```
<a target="_blank" rel="noopener noreferrer" href="javascript:alert('cookies'+'://')">test</a>
```

Reply:
@ssisk @wandb-kc as far as I know, I would not expect the source URLs to come from end users, but we don't have checks to stop devs from doing that right now. 

I cannot reuse the same `getSafeUrlWithoutXss` function because it is valid to pass in a path without http:// or https:// because we use this component to also go to a different path within the app. fwiw, i've kept existing functionality and just added `mailto` to the check. 

Example:
```
<TargetBlank href="/fc">Go to Fully Connected →</TargetBlank>
```

Reply:
`<a href="mailto:sales@wandb.com">sales@wandb.com</a>` is also all over core app. Would this be less of a risk?

Reply:
Ah got it. In that case, as a follow-up to this PR, I think we should rename `TargetBlank` to `UnsanitizedTargetBlank` and add a comment to warn against using the component with untrusted hrefs. If your team doesn't have the bandwidth, then let me know and we will follow up with the PR and request a stamp.

`<a href="mailto:sales@wandb.com">sales@wandb.com</a>` should be safe if it is hardcoded

Reply:
TargetBlank is kind of a misnomer if you are using it for a mailto. I think it would be preferable to have a component like `<Mailto email="sales@wandb.com" />` and just not allow mailto protocol in TargetBlank.

Reply:
I like Jamie's proposal, so going to close this PR and open a new one to keep things clean